dash/include/dash/algorithm/Copy.h

```diff
@@ -493,17 +493,42 @@ GlobOutputIt copy_impl(
                  "l_in_last:",   in_last,
                  "g_out_first:", out_first.pos());
 
+
+
   auto num_elements = std::distance(in_first, in_last);
-  dart_storage_t ds = dash::dart_storage<ValueType>(num_elements);
-  DASH_ASSERT_RETURNS(
-    dart_put_blocking(
-      out_first.dart_gptr(),
-      in_first,
-      ds.nelem,
-      ds.dtype),
-    DART_OK);
 
-  auto out_last = out_first + num_elements;
+  if (num_elements < 1) return out_first;
+
+  auto nremaining = static_cast<dash::default_size_t>(num_elements);
```

Comment:
Dangerous! `dash::default_size_t` may be 32 bit wide while `num_elements` is of type `size_t`.